jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -393,103 +446,109 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
             uncertainty=uncertainty,
             mask=mask
         )
+        return collapsed_spec
+
+    @with_spinner()
+    def extract(self, return_bg=False, add_data=True, **kwargs):
+        """
+        Extract the spectrum from the data cube according to the plugin inputs.
+
+        Parameters
+        ----------
+        return_bg : bool, optional
+            Whether to also return the spectrum of the background, if applicable.
+        add_data : bool, optional
+            Whether to load the resulting data back into the application according to
+            ``add_results``.
+        kwargs : dict
+            Additional keyword arguments passed to the NDDataArray collapse operation.
+            Examples include ``propagate_uncertainties`` and ``operation_ignores_mask``.
+        """
+        if self.conflicting_aperture_and_function:
+            raise ValueError(self.conflicting_aperture_error_message)
+        if self.aperture.selected == self.background.selected:
+            raise ValueError("aperture and background cannot be set to the same subset")
+
+        selected_func = self.function_selected.lower()
+        spec = self._extract_from_aperture(self.spectral_cube, self.uncert_cube,
+                                           self.aperture, self.wavelength_dependent,
+                                           selected_func, **kwargs)
+
+        bg_spec = self.extract_bg_spectrum(add_data=False)
+        if bg_spec is not None:
+            spec = spec - bg_spec
+
+        # per https://jwst-docs.stsci.edu/jwst-near-infrared-camera/nircam-performance/nircam-absolute-flux-calibration-and-zeropoints # noqa
+        pix_scale_factor = self.aperture_area_along_spectral * self.spectral_cube.meta.get('PIXAR_SR', 1.0)  # noqa
```

Comment:
this is technically unrelated to this PR but makes use of the new `aperture_area_along_spectral` implementation here.  The "bug" technically is in 3.10, but is not exposed besides being in the hidden metadata entry, so I don't think needs to be backported or included in the changelog separately - but if anyone thinks otherwise, we can split out the logic needed for this in a manual bugfix backport.

Reply:
cc @gibsongreen

Reply:
ok, I'm removing this from this PR and opening another as a follow-up where we can also address the necessary changes to tests